miso-web/src/main/webapp/scripts/sample_hot.js

```diff
@@ -793,10 +793,11 @@ Sample.hot = {
     return sample;
   },
   
-  saveOneSample: function (data, index, messages, numberToSave) {
+  saveOneSample: function (data, index, messages, numberToSave, callback) {
     var xhr = new XMLHttpRequest();
     xhr.onreadystatechange = function() {
       if (xhr.readyState === XMLHttpRequest.DONE) {
+    	callback(); // Indicate request has completed.
```

Comment:
Yes. At this point the REST request is complete and either ended in success or failure. The next save can be triggered before the successSave or failSave callbacks update the page with the results.